File: php/task2/4mod2.php
<?php
function countDigits($nums) { // Функция для подсчёта чисел 
    $count = 0;
    foreach ($nums as $num) { // Цикл for для массива
        $num = abs($num); 
        $digits = 0;
        if ($num == 0) {
            $digits = 1;
        } else {
            while ($num != 0) {
                $num = intdiv($num, 10); 
                $digits++;
            }
            if ($digits % 2 != 0) {
                $count++;
            }
        }
    }
    return $count;
}

// Main
echo "Введите размер массива:\n";
$size = intval(trim(fgets(STDIN))); // Считываем размер массива
$numbers = array(); // Массив чисел

// Считываем элементы массива
for ($i = 0; $i < $size; $i++) {
    echo ($i + 1) . " Элемент массива: ";
    $numbers[$i] = intval(trim(fgets(STDIN))); // Считываем элементы
}

echo countDigits($numbers) . " Чисел(а) содержат нечетное количество цифр\n";
?>

File: rust/task2/4mod2.rs
use std::io;

fn countDigits(nums: &Vec<i32>) -> i32{ // i32 - то же самое, что и int
    let mut count = 0;
    for &num in nums { // Разыменовываем num (для того чтобы было значение, а не ссылка на значение)
        let mut digits = 0;
        let mut n = num.abs(); // Копия числа
        if n == 0 {
            digits = 1;
        } else {
            while n != 0 {
                n /= 10;
                digits += 1;
            }
        }
        if digits % 2 != 0{
            count += 1;
        }
    }
    return count;
}

fn main(){
    let mut input = String::new();
    println!("Введите рамер массива");

    io::stdin().read_line(&mut input).expect("Ошибка"); // Читаем строку из ввода (ссылка на изменяемую строку), если не получилось - ошибка
    let size: usize = input.trim().parse().expect("Не число"); // trim - убираем пробелы, parse - преобразуем строку в число
    
    input.clear(); // Чистим input

    let mut numbers: Vec<i32> = Vec::new();

    for i in 0..size{
        println!("Введите {} элемент массива", i + 1);
        io::stdin().read_line(&mut input).expect("Ошибка чтения");
        let num: i32 = input.trim().parse().expect("Введите корректное число");
        numbers.push(num);
        input.clear();
    }

    let result = countDigits(&numbers);
    println!("{} числа содержат нечетные числа", result);
}